src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1064,887 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+
+    # Adding helm repo
+    if os.getenv('HELMREPONAME') and os.getenv('HELMREPOURL'):
+        utils.add_helm_repo(kube_config, kube_context)
+
+    # Setting the config dataplane endpoint
+    config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
+
+    # Retrieving Helm chart OCI Artifact location
+    registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
+
+    reg_path_array = registry_path.split(':')
+    agent_version = reg_path_array[1]
+
+    # Set agent version in registry path
+    if connected_cluster.agent_version is not None:
+        agent_version = connected_cluster.agent_version
+        registry_path = reg_path_array[0] + ":" + agent_version
+
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.AgentVersion': agent_version})
+
+    # Get Helm chart path
+    chart_path = utils.get_chart_path(registry_path, kube_config, kube_context)
+
+    cmd_helm_upgrade = ["helm", "upgrade", "azure-arc", chart_path,
+                        "--reuse-values",
+                        "--wait", "--output", "json"]
+    if values_file_provided:
+        cmd_helm_upgrade.extend(["-f", values_file])
+    if kube_config:
+        cmd_helm_upgrade.extend(["--kubeconfig", kube_config])
+    if kube_context:
+        cmd_helm_upgrade.extend(["--kube-context", kube_context])
+    if enable_azure_rbac:
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.enabled=true"])
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.clientId={}".format(aad_client_id)])
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.clientSecret={}".format(aad_client_secret)])
+    if enable_cluster_connect:
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.clusterconnect-agent.enabled=true"])
+    if enable_cl:
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.customLocations.enabled=true"])
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.customLocations.oid={}".format(custom_locations_oid)])
+
+    response_helm_upgrade = Popen(cmd_helm_upgrade, stdout=PIPE, stderr=PIPE)
+    _, error_helm_upgrade = response_helm_upgrade.communicate()
+    if response_helm_upgrade.returncode != 0:
+        if ('forbidden' in error_helm_upgrade.decode("ascii") or 'timed out waiting for the condition' in error_helm_upgrade.decode("ascii")):
+            telemetry.set_user_fault()
+        telemetry.set_exception(exception=error_helm_upgrade.decode("ascii"), fault_type=consts.Install_HelmRelease_Fault_Type,
+                                summary='Unable to install helm release')
+        raise CLIError(str.format(consts.Error_enabling_Features, error_helm_upgrade.decode("ascii")))
+
+    return str.format(consts.Successfully_Enabled_Features, features, connected_cluster.name)
```

Comment:
arpit, handle the case where we enable cluster-connect along with custom-locations even if user hasnt passed cluster-connect from command line.

Reply:
Yepp, missed it

Reply:
https://github.com/Azure/azure-cli-extensions/pull/3001#discussion_r595048673

Reply:
For disable, my comment wasnt valid as we are erroring out, but for enable a change was required. I have updated the features list.